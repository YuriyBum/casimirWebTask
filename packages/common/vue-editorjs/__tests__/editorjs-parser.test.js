// eslint-disable-next-line import/no-extraneous-dependencies
import serializer from 'jest-serializer-html';
import EditorjsParser from '../lib/parser/editorjs-parser';

expect.addSnapshotSerializer(serializer);

describe('editorjs-parser', () => {
  const parser = new EditorjsParser();

  it('should throw error', () => {
    expect(() => {
      parser.parse('text');
    }).toThrow();
  });

  it('should return empty string', () => {
    const result = parser.parse(null);
    expect(result).toEqual('');
  });

  it('should return parsed html', () => {
    const editorData = {
      time: 1629376009755,
      blocks: [
        {
          id: 'OipISyQbWZ',
          type: 'header',
          data: {
            text: 'header',
            level: 2
          }
        },
        {
          id: '6Uz542s55a',
          type: 'paragraph',
          data: {
            text: 'text'
          }
        }
      ],
      version: '2.22.2'
    };

    const result = parser.parse(editorData);
    expect(result).toMatchSnapshot();
  });
});
