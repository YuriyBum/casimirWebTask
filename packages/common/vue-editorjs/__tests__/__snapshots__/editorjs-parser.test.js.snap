// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editorjs-parser should return parsed html 1`] = `
<div class="de-block">
  <h2 class="de-header">
    header
  </h2>
</div>
<div class="de-block">
  <p class="de-paragraph">
    text
  </p>
</div>
`;
